networkx/algorithms/bridges.py

```diff
@@ -0,0 +1,179 @@
+    We formally define a local bridge to be any edge `(u, v)` such that the
+    removal of the edge results in a distance strictly greater than 2 between
+    nodes `u` and `v`. We formally define *span* as the distance between two
+    nodes `u` and `v` when the edge connecting them is removed. Note that all
+    bridges are local bridges with span of infinity (outputs span of
+    float('inf') in this function).
+
+    Parameters
+    ----------
+    G : Undirected Graph object
+
+    Returns
+    ----------
+    iterable
+        Yields a tuple for each local bridge found.
+        Format : `(u, v, d)`
+        `u, v` : edge nodes for this local bridge
+        `d` : span of this local bridge
+
+    Examples
+    --------
+    >>> G = nx.cycle_graph(5)
+    >>> bridges_list = local_bridges(G)
+    >>> list(bridges_list)
+    [(0, 1, 4), (0, 4, 4), (1, 2, 4), (2, 3, 4), (3, 4, 4)]
+
+    Notes
+    ----------
+    This function can be useful to determine what local bridges exist
+    in a given network and what their spans are. The function finds the local
+    bridges as follows. For each edge `(u, v)`, determine if its a local bridge
+    by checking for no intersection in the neighbors of `u` and `v`. If the
+    edge is a local bridge, determine its span by removing `(u, v)` from the
+    graph, computing the new distance between `u` and `v` and then adding the
+    edge back. If removal of the edge results in an impossible path, we have
+    a bridge. Standard convention is to assign infinite span to bridges.
+    """
+
+    H = G.copy(with_data=False)
+    for u, v in G.edges():
+        if not (set(G[u]) & set(G[v])):
+            # edge has a span of at least 2. this is a local bridge
+            H.remove_edge(u, v)
+            try:
+                d = nx.shortest_path_length(H, u, v)
+            except NetworkXNoPath:
+                d = float('inf')
+            H.add_edge(u, v)
+            yield u, v, d
+
+
+def bridges(G):
+    """ Looks through the graph object `G` for all bridges.
+
+    We formally define a bridge to be any edge `(u, v)` such that the removal
+    of the edge increases the total number of connected components.
+
+    Parameters
+    ----------
+    G : Undirected Graph object
+
+    Returns
+    ----------
+    iterable
+        Edges that are bridges
+
+    Examples
+    --------
+    >>> G = nx.cycle_graph(5)
+    >>> bridges_list = bridges(G)
+    >>> list(bridges_list)
+    []
+    >>> G.remove_edge(0,1)
+    >>> list(bridges(G))
+    [(1, 2), (2, 3), (3, 4), (0, 4)]
+
+    Notes
+    ----------
+    This function can be useful to quickly determine what bridges exist
+    in a given network. We first construct the biconnected components
+    of the graph. Then, since any bridge must exist only in a two-node
+    biconnected component, we simply iterate and output the two-node
+    sets. This iteration is a linear time operation.
+    """
+
+    biconnects = nx.biconnected_components(G)
```

Comment:
This should be
```python
for component in nx.biconnected_components(G):
    if len(component) == 2:
        yield tuple(component)
```
without the initial blank line in the body of the function.

Why? I use `component` as the variable of iteration over each biconnected component instead of `bridge_pair`, since the variable does not necessarily hold set of cardinality two on each iteration. I use `yield tuple(component)` because we already know the component is a set of cardinality two, so expressed as a tuple, the component is itself an edge.